substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -763,6 +771,52 @@ pub mod pallet {
 			use frame_support::traits::CallerTrait;
 			o.as_signed().and_then(Self::rank_of)
 		}
+
+		/// Ensure the correctness of the state of this pallet.
+		#[cfg(any(feature = "try-runtime", test))]
+		pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+			Self::try_state_members()?;
+			Self::try_state_index()?;
+
+			Ok(())
+		}
+
+		/// ### Invariants of Member storage items
+		///
+		/// Total number of [`Members`] in storage should be >= [`MemberIndex`] of a [`Rank`] in
+		///    [`MemberCount`].
+		#[cfg(any(feature = "try-runtime", test))]
+		fn try_state_members() -> Result<(), sp_runtime::TryRuntimeError> {
+			MemberCount::<T, I>::iter().try_for_each(|(_rank, member_index)| -> DispatchResult {
+				let total_members = Members::<T, I>::iter().count();
+				ensure!(
+				total_members as u32 >= member_index,
```

Comment:
Is this >= or ==?

Reply:
This is >= because total_members in storage is always >= member_index

Reply:
I think a better invariant to check here is that total_members equals the sum of all member_indexes?

Reply:
Ah ok, thanks for explaining. 

In this case, can we calculate what the index should be for each rank based on members?

e.g. if we have 1 member in rank 1 and one in rank 3, MemberCount should be

0: 2
1: 1
2: 1
?

Do I understand that right?

Reply:
But then MemberCount actually already does this since it holds the storage of rank to index